src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -0,0 +1,376 @@
+//! All icons that are used in the searcher.
+
+use ensogl::prelude::*;
+
+use ensogl::application::Application;
+use ensogl::data::color;
+use ensogl::display::DomSymbol;
+use ensogl::display::navigation::navigator::Navigator;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::shape::*;
+use ensogl::system::web;
+use ensogl_theme::application::searcher::icons as theme;
+use ensogl::system::web::StyleSetter;
+use std::f32::consts::PI;
+use wasm_bindgen::prelude::*;
+
+/// The width and height of all icons.
+const ICON_SIZE : f32 = 16.0;
+
+// The following constant exists for development purposes only.
+// Due to a rendering error, shapes appear too big when the camera is zoomed in very closely.
+// (Documented here: https://github.com/enso-org/ide/issues/1698)
+// In the user interface, this is not a big problem, since icons are usually shown at lower zoom
+// levels. But it is a problem during development of icons when it becomes necessary to inspect them
+// closely. In those situations, one can apply `.shrink(0.35)` to shapes to compensate for the bug
+// and make them appear at the correct size while the camera is zoomed in. But that work-around will
+// make them appear too thin on the default zoom level.
+//
+// To make it easy to turn this shrinking on and off before and after working on icons, we define
+// the constant `SHRINK_AMOUNT` and apply `.shrink(SHRINK_AMOUNT.px())` to all icons. In every
+// commit, `SHRINK_AMOUNT` should be set to 0.0 to make icons look best in the user interface. But
+// during work on the icons, it can temporarily be set to 0.35.
+const SHRINK_AMOUNT : f32 = 0.35;
+
+
+
+// =============
+// === Arrow ===
+// =============
```

Comment:
I'd rename it to "Shape Components" and I'd put here all functions used to build other components, like arrow or cursor. This way it will be named the same way as the next section, which is "IO" and contains all icons belonging to i/o